bears/vcs/VCSCommitBear.py

```diff
@@ -0,0 +1,144 @@
+import abc
+import re
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.HiddenResult import HiddenResult
+from coalib.misc.Shell import run_shell_command
+from coala_utils.decorators import (enforce_signature, generate_ordering,
+                                    generate_repr)
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'raw_commit_message',
+               'commit_sha',
+               'commit_type',
+               'modified_files',
+               'added_files',
+               'deleted_files'
+               )
+@generate_ordering('raw_commit_message',
+                   'commit_sha',
+                   'commit_type',
+                   'modified_files',
+                   'added_files',
+                   'deleted_files',
+                   'severity',
+                   'confidence',
+                   'origin',
+                   'message_base',
+                   'message_arguments',
+                   'aspect',
+                   'additional_info',
+                   'debug_msg')
+class CommitResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin,
+                 raw_commit_message: str,
+                 commit_sha: str,
+                 commit_type: list,
+                 modified_files: list,
+                 added_files: list,
+                 deleted_files: list):
+
+        super().__init__(origin, '')
+        self.raw_commit_message = raw_commit_message
+        self.commit_sha = commit_sha
+        self.commit_type = commit_type
+        self.modified_files = modified_files
+        self.added_files = added_files
+        self.deleted_files = deleted_files
+
+
+class VCSCommitBear(GlobalBear):
+    __metaclass__ = abc.ABCMeta
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    @abc.abstractmethod
+    def get_head_commit(self):
+        """
+        Return the commit message from the head commit
+        """
+
+    def analyze_git_commit(self, head_commit):
+        """
+        Check the current commit at HEAD.
+
+        Return the commit information such as commit message,
+        type of commit, commit SHA, modified, added and
+        delete files by the commit.
+
+        :param head_commit: The commit message at HEAD
+        :return:            A tuple comprising of commit message,
+                            commit sha, type of commit and lists of
+                            modified, added and deleted files
+        """
+        raw_commit_message = head_commit
+
+        commit_sha = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+
+        commit_type = []
```

Comment:
Another good idea for this thing is to use `Flag`. It has several advantages than using this kind of list, such as more little memory usage, the type is constant (which means we can't add unknown `commit_type` in it), and more. See `URLBear.LINK_CONTEXT` for the usage, we already has `aenum` installed in this repo's requirements, so it's worth to use it. It'd be very nice if you can implement it :)